lib/action.ml

```diff
@@ -140,14 +139,14 @@ module Action (Github_api : Api.Github) (Slack_api : Api.Slack) = struct
       | Some (Ignore, _, _) | None -> Lwt.return []
       | Some (Allow, notify_channels, notify_dm) -> action_on_match n.branches ~notify_channels ~notify_dm
       | Some (Allow_once, notify_channels, notify_dm) ->
-      match Map.find repo_state.pipeline_statuses pipeline with
+      match StringMap.find_opt pipeline repo_state.pipeline_statuses with
       | Some branch_statuses ->
-        let has_same_status_state_as_prev (branch : branch) =
-          match Map.find branch_statuses branch.name with
+        let has_different_status_state_from_prev (branch : branch) =
+          not @@ match StringMap.find_opt branch.name branch_statuses with
           | None -> false
-          | Some state -> Poly.equal state current_status
+          | Some state -> state = current_status
         in
-        let branches = List.filter n.branches ~f:(Fn.non @@ has_same_status_state_as_prev) in
+        let branches = List.filter has_different_status_state_from_prev n.branches in
```

Comment:
OCaml has `Fun.negate` so you can avoid slapping a `not` before a match expression.